src/Symfony/Component/HttpKernel/DependencyInjection/AddCacheClearerPass.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Registers the cache clearers.
+ *
+ * @author Dustin Dobervich <ddobervich@gmail.com>
+ */
+class AddCacheClearerPass implements CompilerPassInterface
```

Comment:
Besides of the sort, this pass and the `AddCacheWarmerPass` are rather identical now (and there's probably more).

A lot of times, a compiler pass for tags is used to collect a bunch of references and that's it. What about a more generic class?

Just a brain fart:

```php
class CollectAndInjectTaggedServicePass implements CompilerPassInterface
{
    const NO_FLAGS = 0;
    const SORT_BY_PRIORITY = 1;

    public function __construct($targetService, $serviceArgumentIndex, $tag, $flags = self::NO_FLAGS);
}
```